lib/ble_manager.dart

```diff
@@ -34,9 +34,7 @@ abstract class BleManager {
   static BleManager _instance;
 
   factory BleManager() {
-    if (_instance == null) {
-      _instance = InternalBleManager();
-    }
+    _instance ??= InternalBleManager();
```

Comment:
wow!!

Reply:
Use `pedantic` from the beginning and it can show you quite some tricks 👍